src/game/server/entities/character.cpp

```diff
@@ -469,7 +465,6 @@ bool CCharacter::GiveWeapon(int Weapon, int Ammo)
 void CCharacter::GiveNinja()
 {
 	m_Ninja.m_ActivationTick = Server()->Tick();
-	m_Ninja.m_CurrentMoveTime = -1;
```

Comment:
When the ninja time ends while you're in a ninja move it will automatically finish that move when you pick up a new ninja powerup in the future. Cause the value is not resetted anymore here. Probably best to reset it when the ninja ends.